### ISSUE TITLE
Cannot read properties of undefined (reading 'trim')

### ISSUE THREAD

**Issue by DerBasler** [AzureAD/microsoft-authentication-library-for-js#4277]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.19.0

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

2.0.0-alpha.3

### Description

We saw in our Application Insights the following error:
**Cannot read properties of undefined (reading 'trim')**

Now I am unsure where to fix the issue.
A quick-fix would be to: 
```
    StringUtils.trimArrayEntries = function (arr) {
        return arr.map(function (entry) { return entry.trim(); });
    };
```
But the issue should probably be fixed when the code tries to get the scope from the cache/localstorage/cookie.

### Error Message

**ERROR** 
ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'trim')
TypeError: Cannot read properties of undefined (reading 'trim')
    at StringUtils.js:89
    at Array.map (<anonymous>)
    at Function.push../node_modules/@azure/msal-common/dist/utils/StringUtils.js.StringUtils.trimArrayEntries (StringUtils.js:89)
    at new ScopeSet (ScopeSet.js:22)
    at RequestParameterBuilder.push../node_modules/@azure/msal-common/dist/request/RequestParameterBuilder.js.RequestParameterBuilder.addScopes (RequestParameterBuilder.js:39)
    at AuthorizationCodeClient.push../node_modules/@azure/msal-common/dist/client/AuthorizationCodeClient.js.AuthorizationCodeClient.createAuthCodeUrlQueryString (AuthorizationCodeClient.js:276)
    at AuthorizationCodeClient.<anonymous> (AuthorizationCodeClient.js:43)
    at step (_tslib.js:75)
    at Object.next (_tslib.js:56)
    at _tslib.js:49
    at resolvePromise (zone-evergreen.js:798)
    at zone-evergreen.js:705
    at zone-evergreen.js:721
    at ZoneDelegate.invoke (zone-evergreen.js:364)
    at Object.onInvoke (core.js:27437)
    at ZoneDelegate.invoke (zone-evergreen.js:363)
    at Zone.run (zone-evergreen.js:123)
    at zone-evergreen.js:857
    at ZoneDelegate.invokeTask (zone-evergreen.js:399)
    at Object.onInvokeTask (core.js:27425)

**Call Stack**
If the scope contains an undefined element the code crashes when it tries to create a ScopeSet
`var scopeArr = inputScopes ? StringUtils.trimArrayEntries(__spreadArrays(inputScopes)) : [];`
inside the trimArrayEntries function
```
    StringUtils.trimArrayEntries = function (arr) {
        return arr.map(function (entry) { return entry.trim(); });
    };
```

If I follow the call stack further I can see that request inside the MsalService already has an undefined scope
```
    loginRedirect(request?: RedirectRequest): Observable<void> {
        return from(this.instance.loginRedirect(request));
    }
```
According to the RedirectRequest type it should contain an ` scopes: Array<string>;`

If I go up again I can see:
```
        this.authService.getLogger().verbose("Guard - logging in by redirect");
        const redirectStartPage = this.getDestinationUrl(url);
        this.authService.loginRedirect(Object.assign({ redirectStartPage }, this.msalGuardConfig.authRequest));
        return of(false);
```
Where `this.msalGuardConfig.authRequest.scopes` contains` [undefined]`. 

Up again I land in the lib\msal-angular\src\msal.guard.ts
```
                    if (!this.authService.instance.getAllAccounts().length) {
                        if (state) {
                            this.authService.getLogger().verbose("Guard - no accounts retrieved, log in required to activate");
                            return this.loginInteractively(state);
                        } 
                       .....
```

### MSAL Configuration

```javascript
//They are correct and everything works
    auth: {
      clientId: environment.msalConfig.clientId,
      authority: environment.msalConfig.instance,
      redirectUri: environment.msalConfig.redirectUri || '/login',
      navigateToLoginRequestUrl: false,
      postLogoutRedirectUri: environment.msalConfig.redirectUri,
    },
    cache: {
      cacheLocation: BrowserCacheLocation.LocalStorage,
      storeAuthStateInCookie: isIE, // set to true for IE 11
    },
```

### Relevant Code Snippets

```javascript
None
```

### Reproduction Steps

1. Login
2. Clearing the sites data (unsing devtools --> Application)
3. Navigating to a new menu point

### Expected Behavior

Should not throw any error. 

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Chrome, Firefox, Edge, Safari, Internet Explorer

### Source

External (Customer)

**Comment by tnorling** [comment]
@DerBasler Can you please upgrade to the GA version of msal-angular and let us know if this is still an issue? If so, can you please also share your MsalGuard config? Thanks!

**Comment by DerBasler** [comment]
@tnorling sorry for the inconvenience the issue seams to be resolved in "@azure/msal-angular": "2.0.5".
